src/test/java/reporter/CustomReporter.java

```diff
@@ -0,0 +1,117 @@
+                List<String> allTestCasesInSuite =
+                        Arrays.stream(tc.getAllTestMethods()).map(ITestNGMethod::getMethodName).toList();
+                List<String> passedTests =
+                        tc.getPassedTests().getAllResults().stream().map(ITestResult::getName).toList();
+                passedTC.put(1, passedTests);
+                List<String> failedTests =
+                        tc.getFailedTests().getAllResults().stream().map(ITestResult::getName).toList();
+                failedTC.put(5, failedTests);
+                List<String> skippedTests =
+                        tc.getSkippedTests().getAllResults().stream().map(ITestResult::getName).toList();
+                skippedTC.put(6, skippedTests);
+                System.out.println("Passed tests for suite '" + suiteName +
+                        "' is:" + passedTests.size());
+                System.out.println("Failed tests for suite '" + suiteName +
+                        "' is:" + failedTests.size());
+                System.out.println("Skipped tests for suite '" + suiteName +
+                        "' is:" + skippedTests.size());
+
+                EyesSlack.sendTestExecutionStatusToSlack(passedTests, failedTests, skippedTests);
+
+                APIClient client = new APIClient(endPoint);
+                client.setUser(username);
+                client.setPassword(password);
+                createTestRailInstance();
+                setProjectSuiteId("TestProject", "Master");
+                createRun();
+
+                JSONObject allTestCases =
+                        (JSONObject) client.sendGet("get_cases/" + projectId + "&suite_id=" + suiteId);
+
+                JSONArray c = (JSONArray) allTestCases.get("cases");
+
+                Map<Long, List<String>> idAndTitles = new HashMap<>();
+
+                for (Object o : c) {
+                    List<String> titles = new ArrayList<>();
+                    JSONObject obj = (JSONObject) o;
+                    Long id = (Long) obj.get("id");
+                    String title = (String) obj.get("title");
+                    if (allTestCasesInSuite.contains(title)) {
+                        titles.add(title);
+                        idAndTitles.put(id, titles);
+                    }
+                }
+
+                getStatusIdAndTcId(passedTC, idAndTitles);
+                getStatusIdAndTcId(failedTC, idAndTitles);
+                getStatusIdAndTcId(skippedTC, idAndTitles);
+
+
+                statusIdAndCaseId.forEach((key, value) -> updateRun(value));
+                statusIdAndCaseId.forEach((key, value) -> addResult(idAndComment.get(key),
+                        Integer.parseInt(value.toString())));
+                statusIdAndCaseId.forEach((key, value) -> addStatusForCase(key, Integer.parseInt(value.toString())));
+            }
+        }
+    }
+
+    private void getStatusIdAndTcId(Map<Integer, List<String>> tests, Map<Long, List<String>> idAndTitles) {
```

Comment:
I guess the method name 'get' is incorrect because it returns nothing

Reply:
fixed